java/server/src/org/openqa/selenium/remote/server/rest/ResultConfig.java

```diff
@@ -166,6 +168,34 @@ public Response handle(Command command) throws Exception {
     return response;
   }
 
+  private String removeCredentials(String command) {
+    if (Strings.isNullOrEmpty(command)) {
+      return command;
+    }
+
+    if (command.contains("send keys")) {
+      java.util.regex.Pattern credentialsPattern =
+        java.util.regex.Pattern
+          .compile("(.*)(send keys:).*(])", java.util.regex.Pattern.CASE_INSENSITIVE);
+      Matcher matcher = credentialsPattern.matcher(command);
+
+      if (matcher.matches()) {
+        return matcher.group(1) + matcher.group(2) + " REDACTED" + matcher.group(3);
+      }
+    } else {
+
+      java.util.regex.Pattern credentialsPattern =
+        java.util.regex.Pattern
+          .compile("(.*)(https?://).*@(.*$)(.*)", java.util.regex.Pattern.CASE_INSENSITIVE);
```

Comment:
Why do we need the final group (.*). The one preceding that has a '$' which means the string is terminated by then.

Reply:
You're right. We dont actually need a 4th group. I'll remove it.

As a sidenote, this is a good example of how the matcher would just return an empty string rather than null if it didnt match anything